app/src/main/java/com/ncnf/map/MarkerInfoWindowManager.java

```diff
@@ -0,0 +1,78 @@
+package com.ncnf.map;
+
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.maps.android.clustering.ClusterItem;
+import com.google.maps.android.clustering.ClusterManager;
+import com.ncnf.R;
+import com.ncnf.event.Event;
+import com.ncnf.event.EventActivity;
+import com.ncnf.feed.ui.EventAdapter;
+
+import java.util.List;
+
+import static com.ncnf.Utils.UUID_KEY;
+
+public class MarkerInfoWindowManager implements GoogleMap.InfoWindowAdapter, ClusterManager.OnClusterItemClickListener<Marker>, ClusterManager.OnClusterItemInfoWindowClickListener<Marker> {
+
+    private final Context context;
+    private final View window;
+    private Marker item;
+
+    public MarkerInfoWindowManager(Context context){
+        this.context = context;
+        this.window = LayoutInflater.from(context).inflate(R.layout.map_marker_info, null);
+    }
+
+    @Override
+    public boolean onClusterItemClick(Marker item) {
+        this.item = item;
+        Log.i("MIWM", "Clicked on cluster item");
```

Comment:
You should add your log tag to utilities and create a static variable like the DEBUG_TAG

Reply:
You're right, it's cleaner. I modified to use the DEBUG_TAG as those lines are here for debugging purposes (maybe I could even get rid of them, I don't know)